### PR TITLE
config: remove dup code for setting port

### PR DESCRIPTION
**Pull request by dougsland**
This is initial patch to simplify and unify the
configuration management.

GitHub-Issue: https://github.com/containers/hirte/issues/301

### CODE REVIEW

File: src/libhirte/common/cfg.c [hunk 1]
```diff
@@ -9,8 +9,13 @@
 #include <unistd.h>
 
 #include "libhirte/common/list.h"
+#include "libhirte/common/parse-util.h"
 #include "libhirte/hashmap/hashmap.h"
 #include "libhirte/ini/ini.h"
+#include "libhirte/log/log.h"
+
+#include "agent/agent.h"
```
Comment (engelmi) [inline src/libhirte/common/cfg.c:17]: The dependency should be only `agent -> cfg` - this adds a dependency cycle. Same for the manager.

File: src/libhirte/common/cfg.c [hunk 2]
```diff
@@ -91,6 +96,32 @@ void cfg_dispose(struct config *config) {
         free(config);
 }
 
+bool cfg_set_port(enum NodeType node, void *instance, const char *port_s) {
```
Comment (engelmi) [inline src/libhirte/common/cfg.c:99]: I am against a function that uses an enum to determine the type of a second parameter for casting - this is quite brittle, unclear from a caller point of view and not really extensible (although that is probably never going to happen).

Comment (engelmi) [inline src/libhirte/common/cfg.c:99]: Based on the naming `cfg_set_port` it suggests that this function sets the port on the config, but it actually sets it on the passed in agent/manager instance.

File: src/agent/agent.c [hunk 3]
```diff
@@ -584,10 +572,8 @@ bool agent_parse_config(Agent *agent, const char *configfile) {
         }
 
         value = cfg_get_value(agent->config, CFG_MANAGER_PORT);
-        if (value) {
```
Comment (engelmi) [inline src/agent/agent.c:587]: This checks for `NULL` as well as an empty string as far as I know - this would be missing in `cfg_set_port`.

**Review by engelmi — CHANGES_REQUESTED** [review]
I don't think the code for setting the port value should be combined for agent and manager. Keeping it separated is easier (to use) and less error prone. 

Based on the first point in #301, its probably best to provide a dedicated function (or two - for agent and manager - e.g. `cfg_set_default_values_agent`) for setting the default values that is called right after `cfg_initialize`. Then the cfg flow is roughly like this:
1. initialize cfg
2. set default values (either for agent or manager)
3. load configs and override default values

This way we set the default values from the start and not later on when we figure out setting xyz is missing. In addition, there is a central location for it instead of spreading it out. The values themselves, e.g. `HIRTE_DEFAULT_PORT` can still be defined in `protocol.h`.

File: src/agent/agent.c (hunk 3 above)
Comment (dougsland) [reply src/agent/agent.c:587]: we do the validation in the cfg_set_port

File: src/libhirte/common/cfg.c (hunk 1 above)
Comment (dougsland) [reply src/libhirte/common/cfg.c:17]: we need the headers :-/ Decouple the headers in a common area would be nice.

File: src/libhirte/common/cfg.c (hunk 2 above)
Comment (dougsland) [reply src/libhirte/common/cfg.c:99]: Probably other way would be using some field in structure like.

Comment (dougsland) [reply src/libhirte/common/cfg.c:99]: I thought the same, we could change that too but looks like we will drop this patch.

**Comment by dougsland** [comment]
I will do a different approach. Closing for now.

**Pull request closed without merging**